domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/casetype/CaseTypeServiceImplTest.java

```diff
@@ -389,6 +398,34 @@ private EventEntity eventEntity(int order) {
 
     }
 
+    @Nested
+    class HasCaseType {
+
+        private static final String CASE_TYPE_REFERENCE = "TestAddressBookCase";
+        private static final String JURISDICTION_REFERENCE = "TEST";
+
+        @Test
+        @DisplayName("Should return a present result if case type for jurisdiction other then given exist")
+        void shouldCallMapperAndReturnResult_whenRepositoryReturnsAnEntity() {
+            when(caseTypeRepository.caseTypeExists(CASE_TYPE_REFERENCE, JURISDICTION_REFERENCE)).thenReturn(2);
+
+            Optional<Boolean> caseTypeExists = classUnderTest.caseTypeExists(CASE_TYPE_REFERENCE, JURISDICTION_REFERENCE);
```

Comment:
is this up to date? how come it's still returning an Optional?

Reply:
Please refresh